### PR TITLE
Add Book - first version

### PR DESCRIPTION
**Pull request by prajdziswet**

### CODE REVIEW

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        [Test]
+        public void AddNewBook()
+        {
+            Library lib = new Library();
+            Author author = lib.AddAuthor("Lev", "Tolstoj");
+            lib.AddBook("226611156", "War and Peace", author);
+
+            Assert.IsTrue(lib.Books[0].ISBN== "226611156");
+            Assert.IsTrue(lib.Books[0].NameBook == "War and Peace");
+            Assert.IsTrue(lib.Books[0].AuthorBook == lib.AddAuthor("Lev", "Tolstoj"));
```
Comment (alex-skiba-postnl) [inline ClassLibrary/ClassLibrary.Test/TestBooks.cs]: It's very wrong to **get** author by calling **Add** method. that's a violation of the development contract, please never do this.
In this specific case you already have `author` variable above, why won't use it?

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        public void AddIncorrectNameBook_Empty()
+        {
+            bool check = false;
+            try
+            {
+                Library lib = new Library();
+                Author author = lib.AddAuthor("Lev", "Tolstoj");
+                lib.AddBook("226611156", "", author);
+            }
+            catch (NullReferenceException)
```
Comment (alex-skiba-postnl) [inline ClassLibrary/ClassLibrary.Test/TestBooks.cs]: Please start using Shouldly. there's a dedicated way to check if specific exception was thrown: https://docs.shouldly.io/documentation/exceptions/throw#shouldthrowfunc

File: ClassLibrary/ClassLibrary/Library.cs [hunk 3]
```diff
@@ -8,6 +8,27 @@ public class Library
     {
         public List<Book> Books= new List<Book>();
 
+        public Book AddBook(String ISBN, String nameBook,Author author)
+        {
+            if (ISBN.IsNullOrWhiteSpace() || nameBook.IsNullOrWhiteSpace() || author == null)
+            {
+                throw new NullReferenceException("Одно из полей книги пусто");
```
Comment (alex-skiba-postnl) [inline ClassLibrary/ClassLibrary/Library.cs]: There's `ArgumentNullException`

File: ClassLibrary/ClassLibrary/Library.cs [hunk 4]
```diff
@@ -8,6 +8,27 @@ public class Library
...
+            }
+
+            Book returnBook = Books?.FirstOrDefault(x => x.ISBN == ISBN);
+
+            if (returnBook != null)
+            {
+                throw new ArgumentException("Данная книга существует");
```
Comment (alex-skiba-postnl) [inline ClassLibrary/ClassLibrary/Library.cs]: Please use English

File: ClassLibrary/ClassLibrary/Library.cs (hunk 3 above)
Comment (alex-skiba-postnl) [reply ClassLibrary/ClassLibrary/Library.cs]: And please use English

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace ClassLibrary.Test
```
Comment (alex-skiba-postnl) [inline ClassLibrary/ClassLibrary.Test/TestBooks.cs:5]: I think you've missed a test case when we add a valid Author which is not in the Library.

**Review by alex-skiba-postnl — CHANGES_REQUESTED** [review]

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs (hunk 5 above)
Comment (prajdziswet) [reply ClassLibrary/ClassLibrary.Test/TestBooks.cs:5]: I think You're wrong
image

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs (hunk 1 above)
Comment (prajdziswet) [reply ClassLibrary/ClassLibrary.Test/TestBooks.cs]: I agree, I just wanted to additionally check that this is one link.

File: ClassLibrary/ClassLibrary/Library.cs (hunk 4 above)
Comment (prajdziswet) [reply ClassLibrary/ClassLibrary/Library.cs]: Всетаки кирилица определяеться =))). Плохо не там где надо (I agree here), но книги могут быть и на кирилице...

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs (hunk 5 above)
Comment (alex-skiba-postnl) [reply ClassLibrary/ClassLibrary.Test/TestBooks.cs:5]: I didn't express it correctly. Should've been "I think you've missed a test case when we add a Book with a valid Author which is not added to the Library."

File: ClassLibrary/ClassLibrary.Test/TestBooks.cs (hunk 1 above)
Comment (alex-skiba-postnl) [reply ClassLibrary/ClassLibrary.Test/TestBooks.cs]: almost always it's better to create a separate test case for it

File: ClassLibrary/ClassLibrary/Library.cs (hunk 4 above)
Comment (alex-skiba-postnl) [reply ClassLibrary/ClassLibrary/Library.cs]: that's true, but figuring out encoding issues will take a lot of time. so let's just do it without cyrillic.

**Review by alex-skiba-postnl — APPROVED** [review]

**Pull request merged**